lib/bloc/blocs/register_bloc.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:injectable/injectable.dart';
+import 'package:structure_flutter/bloc/events/register_event.dart';
+import 'package:structure_flutter/bloc/states/register_state.dart';
+import 'package:structure_flutter/di/injection.dart';
+import 'package:structure_flutter/repositories/account_repository.dart';
+import 'package:structure_flutter/repositories/user_repository.dart';
+
+@injectable
+class RegisterBloc extends Bloc<RegisterEvent, RegisterState> {
+  final _userRepository = getIt<UserRepository>();
+
+  final _accountRepository = getIt<AccountRepository>();
+
+  RegisterBloc(RegisterState initialState) : super(initialState);
+
+  @override
+  Stream<RegisterState> mapEventToState(RegisterEvent event) async* {
+    if (event is RegisterWithCredentials) {
+      yield* _mapRegisterAccountToState(
+        email: event.email.trim(),
+        password: event.password.trim(),
+        name: event.fullName.trim(),
+        imageURL: null,
```

Comment:
remove null param

Reply:
i will use imageURL later for upload the user avatar !